libs/actionbar/data/smithed.actionbar/advancements/impl/__version__/vanilla/bed/clicked_bed.json

```diff
@@ -0,0 +1,139 @@
+{
+  "criteria": {
+    "unoccupied": {
+      "trigger": "minecraft:item_used_on_block",
+      "conditions": {
+        "player": [
+          {
+            "condition": "minecraft:inverted",
+            "term": {
+              "condition": "minecraft:value_check",
```

Comment:
Why use a score check here? Can't you use an `entity_properties` condition that checks the `is_sneaking` value in the `flags`?

Reply:
```json
{
  "condition": "minecraft:entity_properties",
  "entity": "this",
  "predicate": {
    "flags": {
      "is_sneaking": true
    }
  }
}
```

Reply:
The scoreboard objective and the flag actually check different things. `is_sneaking` is true if the player's model is sneaking, meaning it's true when the player is force sneaking under a half slab. However the scoreboard objective is only true when the player is physically sneaking (i.e. they're holding down shift), meaning it's false when the player is force sneaking. The scoreboard objective is required to detect when the player is able to actually interact with the bed.

Reply:
Oh, I had no idea it was different. That's annoying
Good to know!